packages/x-date-pickers/src/DesktopDateTimePicker/DesktopDateTimePicker.tsx

```diff
@@ -36,9 +37,15 @@ const DesktopDateTimePicker = React.forwardRef(function DesktopDateTimePicker<TD
     DesktopDateTimePickerProps<TDate>
   >(inProps, 'MuiDesktopDateTimePicker');
 
+  const thresholdToRenderTimeInASingleColumn =
```

Comment:
I like the 1st step toward abstracting this logic away to avoid duplication.
But we still have 4 elements duplicated:

1. The `thresholdToRenderTimeInASingleColumn` default value
2. The `timeSteps` default value
3. The function call to get `shouldRenderTimeInASingleColumn`
4. The function call to get `views`

Would it make sense to have a single function that returns `thresholdToRenderTimeInASingleColumn`, `shouldRenderTimeInASingleColumn`, `timeSteps` and `views`?

Something like:


```tsx
const timeViewResponse = resolveTimeViews(defaultizedProps)

const props = {
  ...timeViewResponse,
  // ... the other props 
}
```


Or do you think those behaviors will vary a lot across components?
From what I saw they are pretty much the same between `DesktopTimePicker` and `DestkopDateTimePicker`

Reply:
Thanks for a great observation! 👍 
I've moved it into `date-time-utils` in the latest commit.
Could you take a second look? 🤔